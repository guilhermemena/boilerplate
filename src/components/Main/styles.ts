import styled from 'styled-components';

export const Wrapper = styled.main`
  width: 100%;
  height: 100%;
  background-color: hsl(223, 63%, 36%);
  display: flex;
  justify-content: center;
  align-items: center;
`;

export const Title = styled.h1`
  font-size: 3.6rem;
  color: white;
`;
