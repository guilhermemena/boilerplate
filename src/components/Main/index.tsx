import * as S from './styles';

const Main = () => (
  <S.Wrapper>
    <S.Title>Next JS - Boilerplate</S.Title>
  </S.Wrapper>
);

export default Main;
