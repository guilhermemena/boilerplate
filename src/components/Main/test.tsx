import { render, screen } from '@testing-library/react';

import Main from './index';

describe('<Main />', () => {
  it('should render the heading', () => {
    render(<Main />);

    expect(screen.getByRole('heading', { name: /next js/i })).toBeInTheDocument();
  });

  it('should render the colors correctly', () => {
    const { container } = render(<Main />);

    expect(container.firstChild).toHaveStyle({ 'background-color': 'hsl(223, 63%, 36%)' });
  });
});
