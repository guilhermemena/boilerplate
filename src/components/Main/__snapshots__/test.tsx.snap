// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  background-color: hsl(223,63%,36%);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  font-size: 3.6rem;
  color: white;
}

<main
  class="c0"
>
  <h1
    class="c1"
  >
    Next JS - Boilerplate
  </h1>
</main>
`;
